app/src/main/java/org/groundplatform/android/data/local/room/stores/RoomLocationOfInterestStore.kt

```diff
@@ -48,9 +53,26 @@ class RoomLocationOfInterestStore @Inject internal constructor() : LocalLocation
     locationOfInterestDao.countByDeletionState(surveyId, EntityDeletionState.DEFAULT)
 
   override fun getValidLois(survey: Survey): Flow<Set<LocationOfInterest>> =
-    locationOfInterestDao.getByDeletionState(survey.id, EntityDeletionState.DEFAULT).map {
-      toLocationsOfInterest(survey, it)
-    }
+    locationOfInterestDao
+      .getByDeletionState(survey.id, EntityDeletionState.DEFAULT)
+      .map { toLocationsOfInterest(survey, it) }
+      .map { lois ->
+        // Filter out LOIs with invalid/empty geometries to prevent crashes
+        lois
+          .filter { loi ->
+            val isValid =
+              when (val geometry = loi.geometry) {
+                is Polygon -> geometry.shell.coordinates.isNotEmpty()
+                is MultiPolygon -> geometry.polygons.all { it.shell.coordinates.isNotEmpty() }
+                else -> true // Point, LineString, LinearRing are always valid if they exist
+              }
```

Comment:
Let's add `isValid()` method to `Geometry` class and override it for each type of geometry. That would keep the core logic within the specific geometry implementation class